src/LayerEditor/ui/data/block_item.py

```diff
@@ -0,0 +1,142 @@
+from LayerEditor.ui.data.interface_node_set_item import InterfaceNodeSetItem
+from LayerEditor.ui.data.interpolated_node_set_item import InterpolatedNodeSetItem
+from LayerEditor.ui.data.layer_item import LayerItem
+from LayerEditor.ui.data.region_item import RegionItem
+from LayerEditor.ui.diagram_editor.graphics_items.gs_point import GsPoint
+from LayerEditor.ui.diagram_editor.graphics_items.gs_polygon import GsPolygon
+from LayerEditor.ui.diagram_editor.graphics_items.gs_segment import GsSegment
+
+from LayerEditor.ui.tools.id_map import IdMap, IdObject
+from LayerEditor.ui.tools.selection import Selection
+
+from gm_base.geometry_files.format_last import InterfaceNodeSet, NodeSet, InterpolatedNodeSet, StratumLayer, \
+    FractureLayer
+
+
+class BlockItem(IdObject):
+    """Holds common data for a block of layers"""
+    def __init__(self, regions_model):
+        super(BlockItem, self).__init__()
+        """Reference for LEData."""
+        self.regions_model = regions_model
+        """Reference to object which manages regions."""
+        self.layers = []
+        """list of layer in this block"""
+        self.layers_dict = IdMap()
+
+        self.selection = Selection()
+        """Selection is common for all layers in this block."""
+
+        self.gui_selected_layer = None
+        """Currently active layer. Region changes are made on this layer."""
+
+
```

Comment:
`decomposition` changed since decomposition is defined by `InterfaceNodeSetItem`